### ISSUE TITLE
Some NZB Sites starting to block SickRage due to API Issues

### ISSUE THREAD

**Issue by adprom** [SiCKRAGETV/sickrage-issues#1556]
Due to the load that sickrage is putting on some of the nzb sites - there is at least one site that has begun blocking sickrage. There appears to be an issue that only affects sickrage but not sickbeard. As such, the user agent for sickrage has been blocked.

**Comment by fernandog** [comment]
Maybe you are to agressive in search frequency.

**Comment by adprom** [comment]
It's not me personally that has been blocked - they have blocked sickrage because across the board it is too server intensive.

**Comment by fernandog** [comment]
Site?

**Comment by miigotu** [comment]
Remember the phrase 'Pix or it didn't happen' ? Without proof or at least a name for us to check ourselves this is nonsense. Invalid.

**Comment by gts24** [comment]
http://i.imgur.com/2xjJx9Y.png
This is a paste from nzbs.in 
SickRage Banned on NZBs.inThursday, June 18 2015 03:10:19 AM

As you all remember, a few weeks ago the site was very slow. This was due to load on our servers, making the site very slow, and unavailable at times.

After some investigation, we found that members using SickRage were hammering the API, causing a large majority of this load.

Furthermore, SickRage doesn't seem to have any throttling mechanism like other apps do.

Since our primary objective and concern is the responsiveness of NZBs.in, we've decided to block the requests coming from SickRage completely,..(at least until there is some throttling capabilities).

The reason for this news item, is because we've learned that the SickRage developers are actively trying to bypass this ban. They've gone as far as changing the identifier of the app to match SickBeard's, just so that indexers can't tell the difference if a request comes from SickRage or SickBeard. 

According to one of SickRage's developers,
"SR is so popular that we were hogging so many resources that their own app could not get results reliably."

Keep in mind, all the others apps work just fine.

So, it’s been decided. No more Sickrage with NZBs.in. 

And if for some reason you sneaky ninjas get around it again, we’ll simply disabled the access to people hammering the API. 

Therefore, we recommend that people using SickRage switch to Sonarr. And whether you take our recommendation or not, we don't care. The moral of the story is SickRage will no longer work on NZBs.in.

**Comment by miigotu** [comment]
That's so ridiculous. Our rate limiting is 2x as strict as SB. There is some serious hating going on here.

Wouldn't you think that these people would come and file an issue if it were true, with some evidence or statistics or something and try to work towards a solution?

Besides that, "a few weeks ago" I didn't rewrite half of the newznab provider yet, where I doubled the limiting.


'learned that the SickRage developers are actively trying to bypass this ban'

Ofcourse, because I fixed many problems, and doubled the limiting from 15 minutes to 30 minutes between unique urls, and because there has been no dialog so we have no way of notifying any of the concerned parties because WE HAVE NO IDEA THAT YOU BANNED US BESIDES SHIT NOT WORKING. If you have an issue with the way our app works, come discuss it, it will get fixed.

'we’ll simply disabled the access to people hammering the API'
Shouldn't this have been the original step?

'The moral of the story is SickRage will no longer work on NZBs.in'
please.

**Comment by adprom** [comment]
miigotu,
Your response there is half of the problem and it is the same every time I raised an issues. Fairly immature "it's not our issue and this problem doesn't exist".

It is why I shifted from sickrage to Sonarr. The devs on SR are acting rather immaturely and every time an issue is raised rather than logically addressing it getting a very childish response.

You mentioned stuff will get fixed - but when people disagree you just tell them that they are wrong.

You admit that you simply try to bypass the ban because you believe yourself that the problems are fixed so bypassing agent restrictions is acceptable. 

I have changed to Sonarr so couldn't care less but hopefully this post provokes a more reasoned response and some reflection. Some serious reflection - not just "this is unfair and people are hating us". Then went down the conspiracy theory path of "this must be a competitor push" - no, there is a perfectly reasonable explanation that SR is causing issues but you don't want to accept that answer.

By the way - you were told there was an issue - I created this thread and you discarded it as untrue. That was when I gave up on SR. 

FYI I have no affiliation with nzbs.in.

**Comment by gts24** [comment]
I also have zero affiliation with nzbs.in , and have personally helped at least 15 people setup sickrage from scratch on various platforms using various service providers. Frankly I've been an SR evangelist of sorts. 
Someone made the comment "pix or it didn't happen" So I provided "pix" in the form of a screenshot.
I really hope this gets resolved because I've had zero issues at all with SR other than this.

**Comment by Thraxis** [comment]
Any site who has felt SickRage was hammering or abusing their system had the ability to contact the devs both via the forums and via issue tickets here where a dialog could have been started and work taken to ensure that SickRage was working as it should.

**Comment by miigotu** [comment]
Half of the problem is I asked for more information on this and you didn't reply to this for over 2 days. That's why it got closed.

When I found an issue in the code, I reopened this issue, referenced it, and fixed the problem.
Our code is limited identically to sick-beard for newznab providers, with the exception that I also keep the cached results for 30 minutes instead of 15 minutes.

`You admit that you simply try to bypass the ban because you believe yourself that the problems are fixed so bypassing agent restrictions is acceptable.`

It's much more than simply, because I made some serious changes before I did that. 
Yes it is acceptable, because I have no idea who has banned us, or why, or any way of contacting any of them. They have also not provided any information to show why they think our app has a problem, or who they are, or any notice, or we would know who to contact.

'no, there is a perfectly reasonable explanation that SR is causing issues but you don't want to accept that answer.'
If so, they could open dialog with us.

They could provide valuable information, like which url configurations? search with q? search with or without season/ep? or is it downloading of nzb files spam?

Providing no information or no contact is like saying "rewrite your whole app because we won't tell you about any problem, and when we ban you you won't know where to look"

@gts24 yes, thanks for posting it. It would just be so much better if they came to use with their issue/findings directly.
